allegations.py

```diff
@@ -0,0 +1,196 @@
+#!/usr/bin/env python3
+
+import re
+
+import pymysql
+
+# Match cases like 8(a)(1) and 8(a)(1)(A)
+# '(?:\([A-Z]+\))?' is an optional non-matching group for the last (A)
+# Could also just do something like this:
+# '(\d(?:\([0-9a-zA-Z]+\))?) (.+)'
+pat = '(\d\([a-z]+\)\(\d+\)(?:\([A-Z]+\))?) (.+)'
+pat = re.compile(pat)
+
+def parse_allegations(s: str) -> (str, str):
+    lines = (x.strip() for x in s.strip().split('\n'))
+    # Ignore empty lines in case of "\n \n" or "\n\n"
+    lines = (l for l in lines if l != "")
+    matches = ((pat.match(line), line) for line in lines)
+    return [
+            (m.group(1), m.group(2), None) if m is not None
+            else (None, None, line)
+            for m,line in matches
+            ]
+
+
+def process_allegations(cursor, case_row):
+    raw = case_row['allegations_raw']
+    case_id = case_row['id']
+
+    exc = None
+    try:
+        okay = True
+        for result in parse_allegations(raw):
+            code, desc, err = result
+            if err is not None:
+                print(f'\tERROR CASE({case_row["case_number"]}): {err}')
+                okay = False
+                continue
+            # Could executemany() here, but then we wouldn't be able to skip on err
+            cursor.execute('INSERT INTO allegations (case_id, code, description) VALUES (%s, %s, %s);', (case_id, code, desc))
```

Comment:
I think you may want to update `cases.allegations_parse_error` here instead of on error, since the default for that field is true.